### PR TITLE
Added dispatch and stop to the job abstraction for api requests

### PR DESCRIPTION
**Pull request by moredure**

### CODE REVIEW

**Comment by moredure** [comment]
As I see, nomad-ruby not supported because there is missing methods not only for dispatch and stop but for job versions, that's why spec for read job not working fails

File: lib/nomad/api/job.rb [hunk 1]
```diff
@@ -11,6 +11,48 @@ def job
...
+    #   dispatch job based on the given name
+    #
+    # @option [String] :payload_contents
+    #   dispatch job payload file content
+    #
+    # @option [Hash] :meta
+    #   dispatch job meta parameters
+    #
+    # @return [JobDispatch]
+    def dispatch(name, payload_contents, meta, **options)
```
Comment (drewbailey) [inline lib/nomad/api/job.rb:31]: Do you mind changing `payload_contents` to `contents` to match create?

**Review by drewbailey — CHANGES_REQUESTED** [review]
Thanks for this contribution! Do you mind adding integration tests for `dispatch` and `stop`? The Job integration specs can be found here https://github.com/hashicorp/nomad-ruby/blob/master/spec/integration/api/job_spec.rb

File: lib/nomad/api/job.rb (hunk 1 above)
Comment (moredure) [reply lib/nomad/api/job.rb:31]: It was long time ago) Currently I will prefer Golang integration over Ruby one

**Comment by spuder** [comment]
I would really like to see this feature implemented.

**Pull request closed without merging**